context/rxjava2/src/main/java/brave/context/rxjava2/internal/fuseable/TraceContextSubscriber.java

```diff
@@ -26,13 +26,12 @@
   }
 
   @Override public final void onSubscribe(Subscription s) {
-    if (SubscriptionHelper.validate(upstream, s)) {
+    if (Util.validate(upstream, s)) {
       downstream.onSubscribe((upstream = s));
```

Comment:
The upstream subscription should not just be passed along.

Reply:
Also `TraceContextSubscriber` should implement `FlowableSubscriber` instead, otherwise RxJava will add extra boundaries around it and fusion will not work.

Reply:
> Also TraceContextSubscriber should implement FlowableSubscriber instead, otherwise RxJava will add extra boundaries around it and fusion will not work.

I assume I should only do this when downstream is also a FlowableSubscriber correct?

Reply:
You should never encounter a plain `Subscriber` inside operators.

Reply:
I guess I just don't quite understand as there seem to be dozens of types that implement Subscriber and not FlowableSubscriber

Reply:
> there seem to be dozens of types

Which ones?